test/unit/delete_publisher.test.ts

```diff
@@ -0,0 +1,60 @@
+import { connect } from "../../src"
+import { expect } from "chai"
+import { Rabbit } from "../support/rabbit"
+import { randomUUID } from "crypto"
+import { expectToThrowAsync, username, password } from "../support/util"
+
+describe("DeletePublisher command", () => {
+  const rabbit = new Rabbit(username, password)
+  const testStreamName = "test-stream"
+  let publisherRef: string
+
+  beforeEach(async () => {
+    publisherRef = randomUUID()
+    await rabbit.createStream(testStreamName)
+  })
+
+  afterEach(() => rabbit.deleteStream(testStreamName))
+
+  it("can delete a publisher", async () => {
+    const connection = await connect({
+      hostname: "localhost",
+      port: 5552,
+      username,
+      password,
+      vhost: "/",
+      frameMax: 0,
+      heartbeat: 0,
+    })
```

Comment:
I suggest to use `createConnection` function from test/support/fake_data.ts

Reply:
I suggest to open and close the connection in the beforeEach and closeEach because closing the connection as last statement of test, if the test fails, can cause the connection remain pending

Reply:
Thanks for pointing that out. I've made use of the createConnection in fake_data.ts, and open and closed teh connection in before/aftereach in this commit https://github.com/coders51/rabbitmq-stream-js-client/commit/7cbc2d08f70627cf6a7270380a5f4e9019919369